openprovider/apis/params_creator.php

```diff
@@ -25,6 +27,43 @@ public function createParameters(array $args, $client, string $method): array
             $this->createParametersGetDelete($args, $client, $method);
     }
 
+    /**
+     * @param array $args
+     * @return array with formatted domain name by idn converter
+     */
+    public function modifyArgsIfDomainIdn(array $args): array
+    {
+        if (empty($args)) {
+            return $args;
+        }
+
+        if (isset($args['domain']['name']) && isset($args['domain']['extension'])) {
```

Comment:
return args inside each if to reduce complexity and nesting
if (...) {
  ...
  return;
}

if (...) {
  ...
  return;
}